presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/RetryDriver.java

```diff
@@ -226,4 +262,87 @@ void nextAttempt()
             }
         }
     }
+
+    public static final class RetryStats
+    {
+        private final CounterStat success = new CounterStat();
+        private final CounterStat finalFailure = new CounterStat();
+        private final DistributionStat attemptsBeforeFailure = new DistributionStat();
+        private final TimeDistribution failureLatency = new TimeDistribution(MICROSECONDS);
```

Comment:
You use microseconds as a unit, but do round to nanos when you update the metric. How does it work in such case?

Reply:
Yes, this seems to be broken. You can call `getUnit()` on `TimeDistribution` when doing the `add()`

Reply:
I think add() takes nano second input and the unit is just for output conversion. In other places of code, we are adding nanosecond into the time distribution.  Testing in test cluster, I see values populated as 70,000 microseconds, so this should be correct?